packages/components/src/internal/renderers/ExpirationDateColumnRenderer.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { FC, memo } from 'react';
+import { Iterable, Map } from 'immutable';
+
+import classNames from 'classnames';
+
+import { QueryColumn } from '../../public/QueryColumn';
+import { isDateTimeInPast } from '../util/Date';
+
+export interface ExpirationDateColumnRendererProps {
+    col?: QueryColumn;
+    columnIndex?: number;
+    data: Map<any, any> | { [key: string]: any };
+    tableCell?: boolean;
+}
+
+export const ExpirationDateColumnRenderer: FC<ExpirationDateColumnRendererProps> = memo(
+    ({ data, columnIndex, col, tableCell }) => {
+        let expired = false;
+        let displayValue = null;
+        let value = null;
+        if (data) {
+            if (Iterable.isIterable(data)) {
+                value = data.get('value');
+                displayValue = data.get('formattedValue') ?? data.get('displayValue');
+            } else if (data['value']) {
+                value = data['value'];
+                displayValue = data['formattedValue'] ?? data['displayValue'];
+            }
+
+            displayValue = displayValue ?? value;
+        }
+
+        if (value) expired = isDateTimeInPast(value);
+
+        if (tableCell) {
+            return (
+                <td
+                    key={columnIndex ?? col?.name}
+                    className={classNames({
+                        'expired-grid-cell': expired,
+                    })}
+                    style={{ textAlign: col?.align || 'left' } as any}
+                >
+                    <div
+                        className={classNames({
+                            'expired-grid-cell-content': expired,
+                        })}
+                    >
+                        {displayValue}
+                    </div>
+                </td>
+            );
+        }
+
+        return (
+            <span
+                className={classNames({
+                    'expired-form-field': expired,
```

Comment:
this display of the red triangle looks a little odd when rendered in the details panel (i.e. not in a grid cell), but it makes sense for consistency.

Reply:
one thing I did expect when I pulled this down locally and looked at it in the UI was to be able to hover over the red triangle to get some information. I get based on the context of this showing up in the Expiration Date column that this indicates a date that is in the past, but thoughts on a brief hover text explaining that? i.e. something like "This sample has an expiration date that is in the past."

Reply:
Andrew: "I don't think a tooltip would add more information that isn't already present: a specific sample is expired, and the date is shown. What else would we want to communicate to the user here?"

Reply:
@labkey-andrewo I see what you mean, but it still seems a little odd to me to show an icon without some sort of description of what it means. We use this same type of styling / icon in LKS for missing value indicators so I guess I'm just a little worried about premium LKS users thinking this might indicate that MVI type of behavior. Thoughts on just using red text for the date string in the expired case?

Reply:
I think we're getting into Gratuitous Redundancy here. We already (think) we communicate: the state of the sample (expired!) and what date it's expiring. Let's not just use red text here because it actually doesn't seem especially visually distinct and it isn't a pattern we use anywhere else.  There are a bunch of explorations of ideas we discarded here: https://www.figma.com/file/4Dg0Ym1XuqvovkQMT4g9zq/Per-Cell-Highlighting-Pattern?node-id=1%3A3165&t=yH0GjXcMcJi8b6YY-1